x/harvest/keeper/liquidation.go

```diff
@@ -0,0 +1,186 @@
+	for _, coin := range borrowBalances {
+		lData := liqMap[coin.Denom]
+		usdValue := sdk.NewDecFromInt(coin.Amount).Quo(sdk.NewDecFromInt(lData.conversionFactor)).Mul(lData.price)
+		totalBorrowedUSDAmount = totalBorrowedUSDAmount.Add(usdValue)
+	}
+
+	// Validate that the proposed borrow's USD value is within user's borrowable limit
+	if totalBorrowedUSDAmount.LTE(totalBorrowableUSDAmount) {
+		return sdkerrors.Wrapf(types.ErrBorrowNotLiquidatable, "borrowed %s <= borrowable %s", totalBorrowedUSDAmount, totalBorrowableUSDAmount)
+	}
+
+	// Sending coins to auction module with keeper address getting 5% of the profits
+	err := k.SeizeDeposits(ctx, deposits, keeper, k.GetKeeperRewardPercentage(ctx))
+	if err != nil {
+		return err
+	}
+
+	borrow, _ := k.GetBorrow(ctx, borrower)
+	k.DeleteBorrow(ctx, borrow)
+
+	return nil
+}
+
+// SeizeDeposits seizes a list of deposits and sends them to auction
+func (k Keeper) SeizeDeposits(ctx sdk.Context, deposits []types.Deposit, keeper sdk.AccAddress, rewardPercentage sdk.Dec) error {
+	for _, deposit := range deposits {
+		keeperReward := rewardPercentage.MulInt(deposit.Amount.Amount).TruncateInt()
+		keeperCoin := sdk.NewCoin(deposit.Amount.Denom, keeperReward)
+		auctionCoin := sdk.NewCoin(deposit.Amount.Denom, deposit.Amount.Amount.Sub(keeperReward))
+
+		// Send auction amount to liquidation module account
+		err := k.supplyKeeper.SendCoinsFromModuleToModule(ctx, types.ModuleName, types.LiquidatorMacc, sdk.NewCoins(auctionCoin))
+		if err != nil {
+			return err
+		}
+
+		// Send keeper their reward
+		err = k.supplyKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, keeper, sdk.NewCoins(keeperCoin))
+		if err != nil {
+			return err
+		}
+
+		err = k.AuctionDeposit(ctx, deposit, rewardPercentage)
+		if err != nil {
+			return err
+		}
+
+		k.DeleteDeposit(ctx, deposit)
+
+		ctx.EventManager().EmitEvent(
+			sdk.NewEvent(
+				types.EventTypeDepositLiquidation,
+				sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
+				sdk.NewAttribute(types.AttributeKeyDepositor, deposit.Depositor.String()),
+				sdk.NewAttribute(types.AttributeKeyDepositCoins, deposit.Amount.String()),
+			),
+		)
+	}
+	return nil
+}
+
+// AuctionDeposit starts auction(s) for an individual deposit
+func (k Keeper) AuctionDeposit(ctx sdk.Context, deposit types.Deposit, reductionPercentage sdk.Dec) error {
+	mm, _ := k.GetMoneyMarket(ctx, deposit.Amount.Denom)
+	amount := reductionPercentage.MulInt(deposit.Amount.Amount).TruncateInt()
+
+	// Initialize auction variables to avoid reusing storage
+	lot := sdk.NewCoin(deposit.Amount.Denom, mm.AuctionSize)
+	returnAddrs := []sdk.AccAddress{deposit.Depositor}
+	weights := []sdk.Int{sdk.NewInt(100)}
+	debt := sdk.NewCoin("debt", sdk.ZeroInt())
+
+	remainingAmount := amount
+	for remainingAmount.GT(mm.AuctionSize) {
+		_, err := k.auctionKeeper.StartCollateralAuction(ctx, types.LiquidatorMacc, lot, lot, returnAddrs, weights, debt)
```

Comment:
The `lot` should not be passed as both the lot and the maxBid argument. To illustrate a simple example:

```
Deposits: 100 KAVA at $2/KAVA ($200 in collateral)
Borrows:  5 BNB at $32.1 (imagine max 0.8 LTV, this position has an LTV of (160.5/200) = 0.8025 and can be liquidated)
```

The auction should be called with args:

```
Lot: 100 KAVA - keeper reward
Max_Bid: 5 BNB
# Note this is the equivalent of selling $200 in KAVA for $160.5 in BNB, which means that the `lot` is more valuable than the `max_bid` by a factor of (1/currentLTV) = (1/0.8025) = 1.246
```

This gets more complicated as you add multiple collaterals/borrows, but the logic should be roughly:

For each borrow coin, that will become the max_bid for an auction. You want to select enough collateral such that the USD value of the `lot` is greater than the USD value of the `max_bid` by 1/LTV. 

Imagine the more complex case:

```
Deposits: 100 KAVA at $2/KAVA ($200 in collateral), 100 XRP at $1/XRP ($100 in collateral) - total $300 collateral
Borrows:  5 BNB at $32.1, 0.004 BTC at $20000 (imagine max 0.8 LTV for both, this position has an LTV of (240.5/300) = 0.80166 and can be liquidated)
```

You would want to do:

```
for each collateralCoin:
  1. Calculate USD value and store it in a map <denom: usdValue>
for each borrowCoin:
  1. Calculate USD value of the total Borrow, store it in a map <denom: usdValue>

for each item in the borrowCoin <denom: usdValue> map:
   1. Divide by the usdValue by the current LTV (0.80166). 
  2. Select a collateral that has capacity for that much USD value, if possible.
  2a. If possible, start an auction for that `lot` and `max_bid` and update the maps of collateral <denom: usdValue> and borrow <denom: usdValue>
  2b. If not possible, figure out how much `max_bid` of collateral we can auction for each partial collateral, and start auctions, the update maps of collateral <denom: usdValue> and borrow <denom: usdValue>

Reply:
Updated in https://github.com/Kava-Labs/kava/pull/731/commits/b087ab9d480e5e9ef1cee2f9e2070ef760850d85, https://github.com/Kava-Labs/kava/pull/731/commits/1a8b25989b0f734147fc486ce541875bda20d136